src/main/java/uk/gov/di/ipv/cri/common/library/util/ClientProviderFactory.java

```diff
@@ -54,7 +56,13 @@ public ClientProviderFactory() {
         awsRegion = Region.of(System.getenv("AWS_REGION"));
         // AWS SDK CRT Client (SYNC) - connection defaults are in SdkHttpConfigurationOption
         sdkHttpClient = AwsCrtHttpClient.builder().maxConcurrency(100).build();
-        environmentVariableCredentialsProvider = EnvironmentVariableCredentialsProvider.create();
+
+        // Check if started inside a snap start container and use appropriate provider
+        // see https://docs.aws.amazon.com/lambda/latest/dg/snapstart-activate.html
+        awsCredentialsProvider =
+                System.getenv("AWS_CONTAINER_CREDENTIALS_FULL_URI") == null
+                        ? EnvironmentVariableCredentialsProvider.create()
```

Comment:
Are we able to write a unit test for this? To check that when null it creates else it builds?

Reply:
Added tests for the ensuring selection of the awsCredentialsProvider is correct